src/expressions/NamedFunctionRef.ts

```diff
@@ -22,7 +23,8 @@ class NamedFunctionRef extends Expression {
 
 	constructor(
 		functionReference: { localName: string; namespaceURI: string | null; prefix: string },
-		arity: number
+		arity: number,
+		type?: SequenceType
```

Comment:
I don't think this needs to be optional?

Reply:
fair point

Reply:
Fixed this.